src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.cs

```diff
@@ -487,36 +487,26 @@ public bool ValueTextEquals(ReadOnlySpan<char> text)
 
             byte[] otherUtf8TextArray = null;
 
-            Span<byte> otherUtf8Text;
-
             int length = checked(text.Length * JsonConstants.MaxExpansionFactorWhileTranscoding);
-
-            if (length > JsonConstants.StackallocThreshold)
-            {
-                otherUtf8TextArray = ArrayPool<byte>.Shared.Rent(length);
-                otherUtf8Text = otherUtf8TextArray;
-            }
-            else
-            {
-                // Cannot create a span directly since it gets passed to instance methods on a ref struct.
-                unsafe
-                {
-                    byte* ptr = stackalloc byte[length];
-                    otherUtf8Text = new Span<byte>(ptr, length);
-                }
-            }
+            Span<byte> otherUtf8Text = length <= JsonConstants.StackallocThreshold ?
+                stackalloc byte[length] :
```

Comment:
`stackalloc byte[JsonConstants.StackallocThreshold]`?

Reply:
I did a quick perf test and it doesn't look like stackalloc a constant length is always faster, especially if the amount being stackalloc'd would otherwise be small:
``` ini

BenchmarkDotNet=v0.11.3.1003-nightly, OS=Windows 10.0.18362
Intel Core i7-6700 CPU 3.40GHz (Skylake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=3.0.100-preview4-010907
  [Host]     : .NET Core 3.0.0-preview7-27803-11 (CoreCLR 4.700.19.30201, CoreFX 4.700.19.30305), 64bit RyuJIT
  DefaultJob : .NET Core 3.0.0-preview4-27519-01 (CoreCLR 4.6.27519.71, CoreFX 4.7.19.16310), 64bit RyuJIT


```
|             Method | Length |     Mean |     Error |    StdDev |   Median |      Min |      Max | Ratio | RatioSD | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|------------------- |------- |---------:|----------:|----------:|---------:|---------:|---------:|------:|--------:|------------:|------------:|------------:|--------------------:|
|    **StackallocConst** |     **10** | **31.29 ns** | **0.2645 ns** | **0.2474 ns** | **31.35 ns** | **30.84 ns** | **31.67 ns** |  **1.00** |    **0.00** |      **0.0057** |           **-** |           **-** |                **24 B** |
| StackallocVariable |     10 | 21.63 ns | 0.4706 ns | 0.4171 ns | 21.68 ns | 20.93 ns | 22.59 ns |  0.69 |    0.01 |      0.0057 |           - |           - |                24 B |
|                    |        |          |           |           |          |          |          |       |         |             |             |             |                     |
|    **StackallocConst** |    **100** | **30.58 ns** | **0.7182 ns** | **1.0749 ns** | **30.46 ns** | **28.61 ns** | **33.55 ns** |  **1.00** |    **0.00** |      **0.0057** |           **-** |           **-** |                **24 B** |
| StackallocVariable |    100 | 26.47 ns | 0.5018 ns | 0.4694 ns | 26.42 ns | 25.64 ns | 27.45 ns |  0.85 |    0.03 |      0.0057 |           - |           - |                24 B |
|                    |        |          |           |           |          |          |          |       |         |             |             |             |                     |
|    **StackallocConst** |    **200** | **31.91 ns** | **0.7349 ns** | **0.8749 ns** | **31.83 ns** | **30.62 ns** | **33.49 ns** |  **1.00** |    **0.00** |      **0.0057** |           **-** |           **-** |                **24 B** |
| StackallocVariable |    200 | 33.45 ns | 0.6816 ns | 0.6043 ns | 33.54 ns | 31.83 ns | 34.35 ns |  1.06 |    0.04 |      0.0057 |           - |           - |                24 B |


To be fair, I was running the perf test from within the dotnet/performance repo, so skipping zero-init is probably not enabled there.

```C#
[Params(10, 100, 200)]
public int Length;

[BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
[Benchmark(Baseline = true)]
public string StackallocConst()
{
    Span<char> result = Length <= 256 ? stackalloc char[256] : new char[Length];

    result[0] = 'a';
    return result.Slice(0, 1).ToString();
}

[BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
[Benchmark]
public string StackallocVariable()
{
    Span<char> result = Length <= 256 ? stackalloc char[Length] : new char[Length];

    result[0] = 'a';
    return result.Slice(0, 1).ToString();
}
```

Reply:
> To be fair, I was running the perf test from within the dotnet/performance repo, so skipping zero-init is probably not enabled there.

Right, that makes the perf test invalid.

Reply:
Just verified locally, and yep, constant variable stackalloc works just fine :)

``` ini

BenchmarkDotNet=v0.11.3.1003-nightly, OS=Windows 10.0.18362
Intel Core i7-6700 CPU 3.40GHz (Skylake), 1 CPU, 8 logical and 4 physical cores
.NET Core SDK=3.0.100-preview4-010907
  [Host]     : .NET Core 3.0.0-preview7-27803-11 (CoreCLR 4.700.19.30201, CoreFX 4.700.19.30305), 64bit RyuJIT
  Job-EPNWWZ : .NET Core ? (CoreCLR 4.700.19.30201, CoreFX 4.700.19.30501), 64bit RyuJIT

Runtime=Core  Toolchain=CoreRun  

```
|             Method | Length |     Mean |     Error |    StdDev |   Median |      Min |      Max | Ratio | RatioSD | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|------------------- |------- |---------:|----------:|----------:|---------:|---------:|---------:|------:|--------:|------------:|------------:|------------:|--------------------:|
|    **StackallocConst** |     **10** | **1.586 ns** | **0.0624 ns** | **0.0553 ns** | **1.567 ns** | **1.517 ns** | **1.715 ns** |  **1.00** |    **0.00** |           **-** |           **-** |           **-** |                   **-** |
| StackallocVariable |     10 | 2.426 ns | 0.0807 ns | 0.0793 ns | 2.404 ns | 2.341 ns | 2.605 ns |  1.54 |    0.08 |           - |           - |           - |                   - |
|                    |        |          |           |           |          |          |          |       |         |             |             |             |                     |
|    **StackallocConst** |    **100** | **1.416 ns** | **0.0324 ns** | **0.0303 ns** | **1.426 ns** | **1.366 ns** | **1.455 ns** |  **1.00** |    **0.00** |           **-** |           **-** |           **-** |                   **-** |
| StackallocVariable |    100 | 2.453 ns | 0.0572 ns | 0.0535 ns | 2.472 ns | 2.365 ns | 2.522 ns |  1.73 |    0.06 |           - |           - |           - |                   - |
|                    |        |          |           |           |          |          |          |       |         |             |             |             |                     |
|    **StackallocConst** |    **200** | **1.575 ns** | **0.0481 ns** | **0.0450 ns** | **1.587 ns** | **1.503 ns** | **1.660 ns** |  **1.00** |    **0.00** |           **-** |           **-** |           **-** |                   **-** |
| StackallocVariable |    200 | 2.435 ns | 0.0703 ns | 0.0658 ns | 2.427 ns | 2.329 ns | 2.569 ns |  1.55 |    0.05 |           - |           - |           - |                   - |